server/src/main/java/org/opensearch/identity/Subject.java

```diff
@@ -0,0 +1,22 @@
+package org.opensearch.identity;
+
+import java.security.Principal;
+
+/**
+ * An individual, process, or device that causes information to flow among objects or change to the system state.
+ * 
+ * Used to authorize actions inside of the OpenSearch ecosystem.
+ */
+public interface Subject {
+    /**
+     * Get the application-wide uniquely identifying principal
+     * */
+    public Principal getPrincipal();
+
+    /**
+     * Check if the permission is allows for this subject
+     *
+     * @param permission The type of these expression of permissions is TDB
+     * */
+    public PermissionResult isPermitted(final String permission);
```

Comment:
Not sure a `String` is the best type here. I see the type is TBD but why not have a `Permission` interface for this API and internals of that TBD?

Reply:
We can do that, I'll make an update in a future iteration